ui/src/app/pages/homepage/data-access-requirements.tsx

```diff
@@ -470,7 +471,11 @@ export const DataAccessRequirements = fp.flow(withProfileErrorModal)((spinnerPro
 
   // handle the route /nih-callback?token=<token>
   // handle the route /ras-callback?code=<code>
-  const {token, code} = queryParamsStore.getValue();
+  const query = useQuery();
+  // Query params can be string or string[], so the compiler will
```

Comment:
This is sort of lying to the compiler, since if I put `?token=123&token=456` in the URL, I'll get a string array at runtime and this will fail in surprising way. See my other comment on `URLSearchParams` which I think should offer a type-safe way to get the values.